p2p/node.go

```diff
@@ -280,6 +280,41 @@ func New(ctx context.Context, config Config) (*Node, error) {
 	return node, nil
 }
 
+// registerValidators registers all the validators we use for incoming and
+// outgoing GossipSub messages.
+func registerValidators(ctx context.Context, basicHost host.Host, config Config, ps *pubsub.PubSub) error {
+	validators := validatorset.New()
+
+	// Add the rate limiting validator.
+	rateValidator, err := ratevalidator.New(ctx, ratevalidator.Config{
+		MyPeerID:       basicHost.ID(),
+		GlobalLimit:    config.GlobalPubSubMessageLimit,
+		GlobalBurst:    config.GlobalPubSubMessageBurst,
+		PerPeerLimit:   config.PerPeerPubSubMessageLimit,
+		PerPeerBurst:   config.PerPeerPubSubMessageBurst,
+		MaxMessageSize: constants.MaxOrderSizeInBytes,
+	})
+	if err != nil {
+		return err
+	}
+	validators.Add("message rate limiting", rateValidator.Validate)
+
+	// Add the custom validator if there is one.
+	if config.CustomMessageValidator != nil {
+		validators.Add("custom", config.CustomMessageValidator)
+	}
+
+	// Register the set of validators for all topics that we publish and/or
+	// subscribe to.
```

Comment:
So does this mean we perform this validation before we publish messages, and on messages received from a subscription? I'm wondering if custom message validation is necessary on the former? Thoughts?

Reply:
Yes, this is a consequence of how the `pubsub` works and can't be easily changed. Luckily validating messages with `jsonschema` is very fast.

Reply:
Ok, that's fine. Can you please add a comment making this clear so we know that this validation is not strictly necessary but we've decided to do it for now?

Reply:
👍 I added a comment explaining this.